scripts/process_cost_data.py

```diff
@@ -35,8 +36,51 @@
 logger = logging.getLogger(__name__)
 
 
+def overwrite_costs(costs: pd.DataFrame, custom_costs: pd.DataFrame) -> pd.DataFrame:
+    """
+    Apply custom cost modifications to costs data.
+
+    Parameters
+    ----------
+    costs : pd.DataFrame
+        Base cost assumptions.
+    custom_costs : pd.DataFrame
+        Custom cost modifications.
+
+    Returns
+    -------
+    costs : pd.DataFrame
```

Comment:
```suggestion
    pd.DataFrame
```

Here not so relevant, but in general mind that in some cases this method modifies the original `costs` dataframe and returns it, and in other cases it returns a new dataframe:

`custom_costs.empty` -> original dataframe
`len(missing_idx) == 0` -> modified original dataframe
`len(missing_idx) == 0` -> a completely new dataframe